frappe/__init__.py

```diff
@@ -818,21 +818,24 @@ def wrapper_fn(*args, **kwargs):
 	return innfn
 
 
-def only_for(roles: list[str] | str, message=False):
-	"""Raise `frappe.PermissionError` if the user does not have any of the given **Roles**.
+def only_for(roles: list[str] | tuple[str] | str, message=False):
+	"""
+	Raises `frappe.PermissionError` if the user does not have any of the permitted roles.
+
+	:param roles: Permitted role(s)
+	"""
 
-	:param roles: List of roles to check."""
-	if local.flags.in_test:
+	if local.flags.in_test or local.session.user == "Administrator":
 		return
 
-	if not isinstance(roles, (tuple, list)):
+	if isinstance(roles, str):
 		roles = (roles,)
-	roles = set(roles)
-	myroles = set(get_roles())
-	if not roles.intersection(myroles):
+
+	if not set(roles).intersection(get_roles()):
 		if message:
 			msgprint(
-				_("This action is only allowed for {}").format(bold(", ".join(roles))), _("Not Permitted")
+				_("This action is only allowed for {}").format(bold(", ".join(roles))),
```

Comment:
I know you didn't make the change now but looking at this, I wonder if telling people which roles are permitted is really ok 

Should we instead just say they don't have permission to access this - and maybe hint at roles while we're at it? 🤔 

With the current message, I have translation concerns too. Wonder if this easily translates to other languages maintaining grammar & clarity.

---

Also, `frappe.throw`

Reply:
>  I wonder if telling people which roles are permitted is really ok

It may help a user with less permissions to approach admin seeking specified role. In any case, the permitted roles aren't _secret_ as long as the source code isn't 😅 

> Should we instead just say they don't have permission to access this

This is the default behaviour. The `message` param is `False` by default.

Reply:
> With the current message, I have translation concerns too. Wonder if this easily translates to other languages maintaining grammar & clarity.

This is what's great about you, Gavin - you care!
Do you think translating individual roles should solve the issue?

Reply:
> Also, `frappe.throw`

I think `throw` wasn't used since `raise PermissionError` is right below it, but I've used it now 👍🏼

Reply:
> Do you think translating individual roles should solve the issue?

I tweaked it a bit to translate individual roles. Happy to make further improvements.

![Screenshot-2022-08-02-192335](https://user-images.githubusercontent.com/16315650/182392379-6a3540f1-4e94-4fc8-931e-4c11e55404aa.png)

Reply:
Looks good now @sagarvora 🤝. Thanks!